### ISSUE TITLE
Writing an open list and then a footer causes an off by one error

### ISSUE THREAD

**Issue by spinningtopsofdoom** [Datomic/fressian#5]
If you write an open list, end it, and then write a footer the validation logic for the number of bytes fails.  The length reported in the footer and the calculated length are off by one.  Here is a example that produces the error

``` clojure
(require '[clojure.data.fressian :as fress])
(import '[org.fressian.impl ByteBufferInputStream BytesOutputStream InheritanceLookup])
(let [my-os (BytesOutputStream.)
      fw (fress/create-writer my-os) ]
  (do
   (fress/begin-open-list fw)
   (fress/end-list fw)
   (fress/write-footer fw)
   (fress/write-object fw nil)
   (let [ my-is (fress/to-input-stream (.internalBuffer my-os))
         fr (fress/create-reader my-is)]
     [(fress/read-object fr) (fress/read-object fr)])))
```

I believe the cause is in the beginOpenList method of FressianWriter

``` java
public Writer beginOpenList() throws IOException {
        if (0 != rawOut.getBytesWritten())
            throw new IllegalStateException("openList must be called from the top level, outside any footer context.");
        writeCode(Codes.BEGIN_OPEN_LIST);
        rawOut.reset();
        return this;
    }
```

The byte code for the open list is written, then the stream is reset making the reported number of bytes written zero, not one.

**Comment by rschmitt** [comment]
Speaking as a member of the peanut gallery, this is an outstanding bug report and I'm surprised that you haven't gotten any kind of response.

**Comment by stuarthalloway** [comment]
Considering https://github.com/Datomic/fressian/commit/07f9443771aa2f74c41fdd4878c45f280f282c2b as a fix for this.

**Comment by puredanger** [comment]
Fixed in 0.6.6.
